appserver/web/web-naming/src/main/java/org/apache/naming/resources/WebDirContext.java

```diff
@@ -174,7 +174,12 @@ public Object lookup(String name)
         if ((file != null && file.isDirectory()) ||
                 (jfEntry != null && jfEntry.getJarEntry().isDirectory())) {
             WebDirContext tempContext = new WebDirContext(env);
-            tempContext.docBase = name;
+            if (file == null) {
+                tempContext.docBase = name;
+            }
+            else {
```

Comment:
Unnecessary line break too
```suggestion
            } else {
```